### PR TITLE
feat: Audio as a backup media

### PR DESCRIPTION
**Pull request by nlewis84**
This PR sets up Audio to be a valid backup on Weekend Content Items and works in conjunction with [this API PR](https://github.com/ApollosProject/apollos-apps/pull/2258/files)

### CODE REVIEW

**Comment by nlewis84** [comment]
This seems to be working on the api end! See below for before and after shots. When I try to test in the app, I am getting crashing on this query. The data looks well formed to me when I replicate in graphical. @redreceipt I could use some help debugging if you have any margin.
Screen Shot 2021-11-09 at 12 27 33 PM

Before:
Screen Shot 2021-11-09 at 11 26 38 AM

After:
Screen Shot 2021-11-09 at 11 25 17 AM

File: packages/apollos-ui-connected/src/NodeSingleConnected/index.js [hunk 1]
```diff
@@ -87,24 +88,39 @@ const NodeSingleConnectedWithMedia = ({
...
     );
 
-  const mediaSource = hasLivestream
-    ? data.node?.liveStream?.media?.sources[0]
-    : data.node?.videos?.find(({ sources }) => sources.length)?.sources[0];
+  const mediaSource = (() => {
+    if (hasLivestream) {
+      return data.node?.liveStream?.media?.sources[0];
+    } else {
+      // eslint-disable-next-line no-lonely-if
```
Comment (redreceipt) [inline packages/apollos-ui-connected/src/NodeSingleConnected/index.js:114]:
```suggestion
```

File: packages/apollos-ui-connected/src/NodeSingleConnected/index.js [hunk 2]
```diff
@@ -87,24 +88,39 @@ const NodeSingleConnectedWithMedia = ({
...
+      if (hasVideo) {
+        return data.node?.videos?.find(({ sources }) => sources?.length)
+          ?.sources[0];
+      } else {
```
Comment (redreceipt) [inline packages/apollos-ui-connected/src/NodeSingleConnected/index.js:118]:
```suggestion
```

File: packages/apollos-ui-connected/src/NodeSingleConnected/index.js [hunk 3]
```diff
@@ -114,6 +130,7 @@ const NodeSingleConnectedWithMedia = ({
         presentationProps={{
           title: data.node.title,
         }}
+        audioOnly={!!(!hasLivestream && !hasVideo && hasAudio)}
```
Comment (redreceipt) [inline packages/apollos-ui-connected/src/NodeSingleConnected/index.js:133]: what's the purpose of this prop?

Comment (nlewis84) [reply packages/apollos-ui-connected/src/NodeSingleConnected/index.js:133]: It passes as true if only Audio sources exist. https://github.com/ApollosProject/apollos-apps/blob/master/packages/apollos-ui-media-player/src/Container.tsx#L41

I believe that it keeps the placeholder image in the media player while playing the audio.

File: packages/apollos-ui-connected/src/NodeSingleConnected/index.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-else-return */
```
Comment (redreceipt) [inline packages/apollos-ui-connected/src/NodeSingleConnected/index.js:1]:
```suggestion
```

File: packages/apollos-ui-connected/src/NodeSingleConnected/index.js (hunk 3 above)
Comment (redreceipt) [reply packages/apollos-ui-connected/src/NodeSingleConnected/index.js:133]: oh nice

Comment (redreceipt) [inline packages/apollos-ui-connected/src/NodeSingleConnected/index.js:133]:
```suggestion
        audioOnly={(!hasLivestream && !hasVideo && hasAudio)}
```

File: packages/apollos-ui-connected/src/NodeSingleConnected/index.js [hunk 5]
```diff
@@ -87,24 +87,36 @@ const NodeSingleConnectedWithMedia = ({
...
       </NodeSingleConnected>
     );
 
-  const mediaSource = hasLivestream
-    ? data.node?.liveStream?.media?.sources[0]
-    : data.node?.videos?.find(({ sources }) => sources.length)?.sources[0];
+  const mediaSource = (() => {
+    if (hasLivestream) {
+      return data.node?.liveStream?.media?.sources[0];
+      if (hasVideo) {
```
Comment (redreceipt) [inline packages/apollos-ui-connected/src/NodeSingleConnected/index.js:112]:
```suggestion
      if (hasVideo)
```

File: packages/apollos-ui-connected/src/NodeSingleConnected/index.js [hunk 6]
```diff
@@ -87,24 +87,36 @@ const NodeSingleConnectedWithMedia = ({
...
+        return data.node?.videos?.find(({ sources }) => sources?.length)
+          ?.sources[0];
+        return data.node?.audios?.find(({ sources }) => sources?.length)
+          ?.sources[0];
+      }
```
Comment (redreceipt) [inline packages/apollos-ui-connected/src/NodeSingleConnected/index.js:117]:
```suggestion
```

File: packages/apollos-ui-connected/src/NodeSingleConnected/index.js [hunk 7]
```diff
@@ -87,24 +87,36 @@ const NodeSingleConnectedWithMedia = ({
...
       <NodeSingleConnected nodeId={nodeId} Component={Component} {...props}>
         {children}
       </NodeSingleConnected>
     );
 
-  const mediaSource = hasLivestream
-    ? data.node?.liveStream?.media?.sources[0]
-    : data.node?.videos?.find(({ sources }) => sources.length)?.sources[0];
+  const mediaSource = (() => {
+    if (hasLivestream) {
```
Comment (redreceipt) [inline packages/apollos-ui-connected/src/NodeSingleConnected/index.js:110]:
```suggestion
    if (hasLivestream)
```

File: packages/apollos-ui-connected/src/NodeSingleConnected/index.js [hunk 8]
```diff
@@ -87,24 +87,36 @@ const NodeSingleConnectedWithMedia = ({
...
+      return data.node?.liveStream?.media?.sources[0];
+      if (hasVideo) {
+        return data.node?.videos?.find(({ sources }) => sources?.length)
+          ?.sources[0];
+        return data.node?.audios?.find(({ sources }) => sources?.length)
+          ?.sources[0];
+      }
+    }
```
Comment (redreceipt) [inline packages/apollos-ui-connected/src/NodeSingleConnected/index.js:118]:
```suggestion
```

**Comment by nlewis84** [comment]
@redreceipt Okay, this is good for you to look at again.

**Review by redreceipt — APPROVED** [review]

**Pull request merged**